platforms/core-runtime/logging/src/main/java/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandler.java

```diff
@@ -89,33 +77,9 @@ public void featureUsed(final DeprecatedFeatureUsage usage) {
                 error = new GradleException(WARNING_SUMMARY + " " + DefaultGradleVersion.current().getNextMajorVersion().getVersion());
             }
         }
-        if (problemsService != null) {
-            InternalProblemReporter reporter = ((InternalProblems) problemsService).getInternalReporter();
```

Comment:
I may miss something here. Was this the wrong place to do the actual problem report?

Reply:
As far as I remember (it wasn't yesterday when I wrote this), my motivation was to put the problem reporting to the highest point in the hierarchy possible. This location might be right functionality-wise, but I felt it was a more logical, better discoverable place if placed next to the `nag...` calls as possible.